x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceServiceFeature.java

```diff
@@ -15,6 +15,8 @@
  */
 public class ElasticInferenceServiceFeature {
 
+    @Deprecated
     public static final FeatureFlag ELASTIC_INFERENCE_SERVICE_FEATURE_FLAG = new FeatureFlag("eis");
 
+    public static final FeatureFlag INFERENCE_SERVICE_FEATURE_FLAG = new FeatureFlag("elastic_inference_service");
```

Comment:
Just wondering why we prefix the setting with `ELASTIC_` and the feature flag not. Maybe this could lead to confusion considering that we also have generic interfaces named [InferenceService](https://github.com/elastic/elasticsearch/blob/103a8b0960f5542367adf84fde3b3f1896e6c225/server/src/main/java/org/elasticsearch/inference/InferenceService.java#L4).

WDYT?

Reply:
Ah, it was simply because the suggestion was `Feature flag -Des.eis_feature_flag_enabled=true --> -Des.inference.elastic.feature_flag_enabled=true` 

Maybe we change the old one to `DEPRCATRED_EIS_FEATURE_FLAG` and the new one to `ELASTIC_INFERENCE_SERVICE_FEATURE_FLAG`? It's just used in tests and when "enabling" the flag. WDYT?

Reply:
Sounds good!